eth2near/eth2near-block-relay-rs/src/init_contract.rs

```diff
@@ -18,22 +50,23 @@ pub fn init_contract(
         config.eth_requests_timeout_seconds,
         config.state_requests_timeout_seconds,
     );
-    let eth1_rpc_client = Eth1RPCClient::new(&config.eth1_endpoint);
 
-    let start_slot = beacon_rpc_client.get_last_finalized_slot_number().unwrap();
-    let period = BeaconRPCClient::get_period_for_slot(start_slot.as_u64());
+    if init_block_root.is_empty() {
+        init_block_root = beacon_rpc_client.get_checkpoint_root().expect("Fail to get last checkpoint");
+    }
+
+    let eth1_rpc_client = Eth1RPCClient::new(&config.eth1_endpoint);
 
     let light_client_update = beacon_rpc_client
         .get_finality_light_client_update_with_sync_commity_update()
         .unwrap();
-    let block_id = format!(
-        "{}",
-        light_client_update
-            .finality_update
-            .header_update
-            .beacon_header
-            .slot
-    );
+    let finality_slot = light_client_update
+        .finality_update
+        .header_update
+        .beacon_header
+        .slot;
+
+    let block_id = format!("{}", finality_slot);
```

Comment:
Please add more logs on what finality slot is used, the provided `init_block_root`, the snapshot corresponding to this block root, etc.

Reply:
Added for `init_block_root ` in bdf2d6a59bda21a086c609a839d297b166696fa9
The other init data is already printed here https://github.com/aurora-is-near/rainbow-bridge/blob/master/eth2near/contract_wrapper/src/eth_client_contract.rs#L69